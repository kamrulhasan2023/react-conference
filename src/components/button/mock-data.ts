import {
  IFillButton,
  IFillRedirectButton,
  IOutlineButton,
  IOutlineRedirectButton,
} from './interface';

export const buttonProps: IFillButton = {
  label: 'Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  clickHandler: () => {},
  variant: 'fill',
};

export const redirectButtonProps: IFillRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  variant: 'fill',
};

export const outlineButtonProps: IOutlineButton = {
  label: 'Outline Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  clickHandler: () => {},
  variant: 'outline',
};

export const redirectOutlineButtonProps: IOutlineRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  variant: 'outline',
};
