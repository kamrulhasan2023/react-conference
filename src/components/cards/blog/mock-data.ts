import { IBlogCard } from './interface';

export const blogCardProps: IBlogCard = {
  href: 'https://www.lemonhive.com/',
  image: {
    src: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
    alt: 'dummy image 2',
    lqip: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
  },
  title: 'Headless Website Development',
  description:
    'Headless solutions built with Sanity.io, Strapi, Prismic, Contentful and others with front-end frameworks such as Next.js, Gatsby, & Nuxt.',
  tags: [
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Reactjs' },
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Nextjs' },
  ],
  isFeatured: false,
};
