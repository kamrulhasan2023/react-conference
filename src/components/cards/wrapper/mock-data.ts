import { ICardWrapper } from './interface';

export const cardWrapperProps: ICardWrapper = {
  cardBorderRadius: 'radius-none',
  cardShadow: 'shadow-one',
};
