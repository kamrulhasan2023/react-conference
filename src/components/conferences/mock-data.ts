import {IConferences} from "./interface";
export const conferencesProps:IConferences = {
    title : "Conferences",
    conferences: [
        {   
            id:1,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        },
        {
            id:2,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        },
        {
            id:3,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
          startDate: "02 September, 2023",
        },
        {
            id:4,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
           
        },
        {
            id:5,
            title: "Freezing Edge 2023",
            description:"The edge isn't bleeding, it's freezing!",
            startDate: "02 September, 2023",
        }
    ]

}