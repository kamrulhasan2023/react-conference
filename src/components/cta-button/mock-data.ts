import { ICtaButton } from './interface';

export const ctaButtonProps: ICtaButton = {
  label: 'Cta button',
  href: '/',
  colorScheme: 'primary',
  isOpenNewTab: false,
};
