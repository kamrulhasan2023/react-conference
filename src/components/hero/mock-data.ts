import { IRCHero } from './interface';
export const heroProps: IRCHero = {
    title: "React Conference",
    description: "Lorem uis diam turpis quam id fermentum.In quis diam turpis quam id fermentum..id fermentum.In quis diam turpi quam id fermentum.",
    btnText: "Buy Tickets",
};