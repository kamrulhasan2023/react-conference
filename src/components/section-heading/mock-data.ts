import { ISectionHeading } from './interface';

export const sectionDescriptionProps = [
  {
    _key: '0ffdc2e0f0b8',
    _type: 'block',
    children: [
      {
        _key: '9ba941b1390f',
        _type: 'span',
        marks: ['805b62da0516'],
        text: 'Headless website',
      },
      {
        _key: '9a6ae5c5c320',
        _type: 'span',
        marks: [],
        text: ' for an award winning digital marketing agency.',
      },
    ],
    markDefs: [
      {
        _key: '805b62da0516',
        _type: 'link',
        href: 'https://www.google.com',
        linkType: 'externalLink',
      },
    ],
    style: 'normal',
  },
  {
    _key: '4ed7297d9dcc',
    _type: 'block',
    children: [
      {
        _key: 'da46c7671dee',
        _type: 'span',
        marks: [],
        text: '',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
  {
    _key: 'c0b3d76ed804',
    _type: 'block',
    children: [
      {
        _key: '1d4ab47cbb880',
        _type: 'span',
        marks: ['strong'],
        text: 'Lorem ipsum',
      },
      {
        _key: '1b8f25fb6ad2',
        _type: 'span',
        marks: [],
        text: ' dolor ',
      },
      {
        _key: '69ae9b1fb5c7',
        _type: 'span',
        marks: ['em'],
        text: 'sit amet consectetur',
      },
      {
        _key: '2e48bd8c68e6',
        _type: 'span',
        marks: [],
        text: '. Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus ',
      },
      {
        _key: 'c0b66fa9a5e6',
        _type: 'span',
        marks: ['underline'],
        text: 'diam lorem nascetur',
      },
      {
        _key: 'c751dbb05102',
        _type: 'span',
        marks: [],
        text: '. Lorem ipsum dolor sit amet consectetur. Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus ',
      },
      {
        _key: '95e7c8d27766',
        _type: 'span',
        marks: ['strong'],
        text: 'diam lorem nascetu ullam nuwel',
      },
    ],
    markDefs: [],
    style: 'normal',
  },
];

export const sectionHeadingProps: ISectionHeading = {
  subtitle: 'Lemon Hive',
  title: 'What Makes Us Different?',
  description: sectionDescriptionProps,
  colorScheme: 'primary',
  hasBottomSpacing: true,
  isCenter: false,
};
