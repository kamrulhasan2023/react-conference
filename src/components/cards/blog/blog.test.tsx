import { render } from '@testing-library/react';
import BlogCard from '.';
import { blogCardProps } from './mock-data';

it('renders course card correctly', () => {
  const { getAllByRole } = render(<BlogCard {...blogCardProps} />);
  const element = getAllByRole('presentation')[0];
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

it('renders course card correctly with link', () => {
  const { getAllByRole } = render(<BlogCard {...blogCardProps} />);
  const element = getAllByRole('link')[0];
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
