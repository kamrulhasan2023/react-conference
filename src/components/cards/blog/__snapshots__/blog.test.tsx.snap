// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders course card correctly 1`] = `
<div
  class="relative overflow-hidden h-auto card shadow-one radius-one"
  role="presentation"
>
  <a
    class="block h-full"
    href="https://www.lemonhive.com/"
    target="_self"
  >
    <div
      class="group/card relative h-full bg-success/[.2]"
    >
      <div
        class="relative"
      >
        <div
          class="overflow-hidden"
        >
          <img
            alt="dummy image 2"
            class="transition-transform ease-in-out group-hover/card:scale-110"
            data-nimg="1"
            decoding="async"
            height="720"
            loading="lazy"
            src="http://localhost/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1920&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1080&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1920&q=75 2x"
            style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
            width="960"
          />
        </div>
      </div>
      <div
        class="flex flex-col space-y-3 p-5 pt-8"
      >
        <h3
          class="line-clamp-2 flex-1 text-lg font-bold"
        >
          Headless Website Development
        </h3>
        <p
          class="line-clamp-3"
        >
          Headless solutions built with Sanity.io, Strapi, Prismic, Contentful and others with front-end frameworks such as Next.js, Gatsby, & Nuxt.
        </p>
        <div>
          <div
            class="mt-4 flex items-center gap-2"
          >
            <span
              class="inline-flex items-center justify-center px-3 py-1 text-sm font-semibold text-black rounded-full bg-danger"
              role="presentation"
            >
              Reactjs
            </span>
            <span
              class="inline-flex items-center justify-center px-3 py-1 text-sm font-semibold text-black rounded-full bg-danger-light"
              role="presentation"
            >
              Nextjs
            </span>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`renders course card correctly with link 1`] = `
<a
  class="block h-full"
  href="https://www.lemonhive.com/"
  target="_self"
>
  <div
    class="group/card relative h-full bg-success/[.2]"
  >
    <div
      class="relative"
    >
      <div
        class="overflow-hidden"
      >
        <img
          alt="dummy image 2"
          class="transition-transform ease-in-out group-hover/card:scale-110"
          data-nimg="1"
          decoding="async"
          height="720"
          loading="lazy"
          src="http://localhost/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1920&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1080&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fdktw59lc2%2Fimage%2Fupload%2Fv1669267967%2FBrightonSEO%2FCarousel%2Fimage-2_yph8q6.jpg&w=1920&q=75 2x"
          style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          width="960"
        />
      </div>
    </div>
    <div
      class="flex flex-col space-y-3 p-5 pt-8"
    >
      <h3
        class="line-clamp-2 flex-1 text-lg font-bold"
      >
        Headless Website Development
      </h3>
      <p
        class="line-clamp-3"
      >
        Headless solutions built with Sanity.io, Strapi, Prismic, Contentful and others with front-end frameworks such as Next.js, Gatsby, & Nuxt.
      </p>
      <div>
        <div
          class="mt-4 flex items-center gap-2"
        >
          <span
            class="inline-flex items-center justify-center px-3 py-1 text-sm font-semibold text-black rounded-full bg-danger"
            role="presentation"
          >
            Reactjs
          </span>
          <span
            class="inline-flex items-center justify-center px-3 py-1 text-sm font-semibold text-black rounded-full bg-danger-light"
            role="presentation"
          >
            Nextjs
          </span>
        </div>
      </div>
    </div>
  </div>
</a>
`;
